### PR TITLE
add a second Clojure solution

### PR DESCRIPTION
**Pull request by Chase-Lambert**
Contributing my attempt at the challenges. I am curious about possible improvements to Challenge #2 specifically.

### CODE REVIEW

File: challenges/clojure-2/core.clj [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+                             (keyword k)))))
+
+(def lib-data
+  (parse-lib-data file))
+
+
+;; Challenge #1:
+
+(defn get-book-property [lib-data isbn field]
+  (get-in lib-data [:catalog :booksByIsbn isbn field]))
```
Comment (viebel) [inline challenges/clojure-2/core.clj:25]: Well done!

File: challenges/clojure-2/core.clj [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                     authors)]
+    (json/write-str
+      {:title (:title book)
+       :isbn  (:isbn book)
+       :authors authors})))
+
+(defn book-info [lib-data title]
+  (let [title (.toLowerCase title)
+        books (all-books lib-data)
+        matches (for [book books
```
Comment (viebel) [inline challenges/clojure-2/core.clj:50]: I think it's more idiomatic to use `filter` here.

File: challenges/clojure-2/core.clj [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        authors (map (fn [author]
+                       (get-in all-authors [author :name]))
+                     authors)]
+    (json/write-str
+      {:title (:title book)
+       :isbn  (:isbn book)
+       :authors authors})))
+
+(defn book-info [lib-data title]
+  (let [title (.toLowerCase title)
```
Comment (viebel) [inline challenges/clojure-2/core.clj:48]: It's better to use `clojure.string/lower-case`

File: challenges/clojure-2/core.clj [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  (get-in lib-data [:catalog :booksByIsbn isbn field]))
+
+;; (get-book-property lib-data :978-1779501127 :title) ;; "Watchmen"
+
+
+;; Challenge #2:
+
+(defn all-books [lib-data]
+  (->> (get-in lib-data [:catalog :booksByIsbn])
+       (map val)))
```
Comment (viebel) [inline challenges/clojure-2/core.clj:34]: Use `vals`

File: challenges/clojure-2/core.clj [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+                  book)]
+    (map book-info->json matches)))
+
+;; (book-info lib-data "watch") ;; ("{\"title\":\"Watchmen\",\"isbn\":\"978-1779501127\",\"authors\":[\"Alan Moore\",\"Dave Gibbons\"]}")
+
+
+;; Challenge #3:
+
+(defn block-member [lib-data email]
+  (assoc-in lib-data [:userManagement :members email :isBlocked] true))
```
Comment (viebel) [inline challenges/clojure-2/core.clj:62]: Nice!

File: challenges/clojure-2/core.clj [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+;; (block-member lib-data "samantha@gmail.com")
+;; => ... {:userManagement {:members {"samantha@gmail.com" {:isBlocked true}}}})
+    
+
+;; Challenge #4:
+
+;; Clojure std library contains this function: `clojure.set/rename-keys`.
+;; (clojure.set/rename-keys {:a 1 :b 2} {:a :new-a :b :new-b}) ;; {:new-a 1, :new-b 2}
```
Comment (viebel) [inline challenges/clojure-2/core.clj:71]: You found the right function for the job!

File: challenges/clojure-2/core.clj [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+(def example-isbn "978-1779501127")
+(def api-data-example
+  (parse-lib-data (slurp (api-url example-isbn))))
+(def lib-data-example
+  (get-in lib-data [:catalog :booksByIsbn (keyword example-isbn)]))
+
+(defn merge-and-serialize [api-data db-data]
+  (json/write-str
+    (merge api-data db-data)))
```
Comment (viebel) [inline challenges/clojure-2/core.clj:87]: In Clojure , data manipulation is easy!

File: challenges/clojure-2/core.clj [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+;; (merge-and-serialize api-data-example lib-data-example)
+
+
+;; Challenge #6:
+
+;; Clojure std library contains this function: `clojure.data/diff`.
+;; (butlast 
+;;   (clojure.data/diff lib-data (block-member lib-data "samantha@gmail.com")))
```
Comment (viebel) [inline challenges/clojure-2/core.clj:96]: Perfect!

**Review by viebel — APPROVED** [review]

**Comment by Chase-Lambert** [comment]
Thank you for the recommendations. I instituted them into the solutions and recommitted.

**Comment by Chase-Lambert** [comment]
And then did another commit to get rid of the .cpcache and add a .gitignore for that.

**Comment by viebel** [comment]
Well done!

**Pull request merged**